apps/avifenc.c

```diff
@@ -1676,9 +1675,18 @@ int main(int argc, char * argv[])
         goto cleanup;
     }
 
-    // Check again for y4m input (y4m input ignores input.requestedFormat and retains the format in file).
-    if ((image->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_IDENTITY) && (image->yuvFormat != AVIF_PIXEL_FORMAT_YUV444) &&
-        (image->yuvFormat != AVIF_PIXEL_FORMAT_YUV400)) {
+    // Check again for -y auto or for y4m input (y4m input ignores input.requestedFormat and
+    // retains the format in file).
```

Comment:
The original comment may have been written before `-y auto` was supported. A y4m input is not the only reason `image->yuvFormat` may be different from `input.requestedFormat` now. So I updated this comment.